lib/serial_terminal.pm

```diff
@@ -114,7 +114,12 @@ sub login {
     die 'Failed to wait for login prompt' unless wait_serial(qr/login:\s*$/i);
     enter_cmd("$user");
     if (length $testapi::password) {
-        die 'Failed to wait for password prompt' unless wait_serial(qr/Password:\s*$/i);
+        if (!wait_serial(qr/Password:\s*$/i)) {
```

Comment:
I'd recommend 4 changes:
- Set timeout to 5 seconds or less (default `wait_serial` timeout is 90 seconds)
- Wait for username to be echoed back instead of password prompt
- Move the workaround out of the parent condition so that it handles even cases where login is done without password
- Record softfail if the first login attempt fails

Reply:
Alternatively, you could just try to insert 1 second sleep before `enter_cmd("$user");` and see whether the race can still happen.